src/bt_peer_connection.cpp

```diff
@@ -2176,7 +2178,7 @@ namespace libtorrent
 			std::memset(ptr, 0, packet_size - 5);
 			piece_picker const& p = t->picker();
 			int mask = 0x80;
-			for (int i = 0; i < num_pieces; ++i)
+			for (piece_index_t i(0); i != piece_index_t(num_pieces); ++i)
```

Comment:
not a problem in this particular case, but the pattern in general. Since `i` is not constant, it could be possible to increment `i` inside the loop, and since now the condition is `!=`, it will fail to stop.